flink-streaming-java/src/main/java/org/apache/flink/streaming/api/transformations/StreamTransformation.java

```diff
@@ -202,7 +203,17 @@ public int getParallelism() {
 	 * @param parallelism The new parallelism to set on this {@code StreamTransformation}
 	 */
 	public void setParallelism(int parallelism) {
-		Preconditions.checkArgument(parallelism > 0, "Parallelism must be bigger than zero.");
+		checkArgument(parallelism != ExecutionConfig.PARALLELISM_UNKNOWN, "Cannot specify UNKNOWN_PARALLELISM.");
```

Comment:
The `PARALLELISM_UNKNOWN` does not seem to be used anywhere in the codebase, apart from checking against it. Couldn't we remove it? Or am I missing something?